--- ./bundle/cscope_maps/plugin/cscope_maps.vim	2022-09-29 01:28:56.968995708 -0700
+++ ./bundle/cscope_maps/plugin/cscope_maps.vim 2022-09-29 01:28:47.533039369 -0700
@@ -39,9 +39,13 @@
 
     " add any cscope database in current directory
     if filereadable("cscope.out")
+        if !filereadable("/.dockerenv")
+            set nocscopeverbose
         cs add cscope.out  
+        endif
     " else add the database pointed to by environment variable 
     elseif $CSCOPE_DB != ""
+        set nocscopeverbose
         cs add $CSCOPE_DB
     endif
 
